File: tests/fixtures/cpp/mars_climate_orbiter.cpp
// Test fixture: Mars Climate Orbiter unit error.
//
// The MCO disaster occurred because Lockheed Martin's SM_FORCES software
// output thrust impulse in pound-force-seconds (lbf*s), while NASA's
// trajectory software expected newton-seconds (N*s).
//
// This fixture models that scenario. The function compute_impulse returns
// a value in lbf*s, but the caller adds it to a value in N*s.
// Expected result: violation flagged at the addition.

double sm_forces_impulse = 100.0;  // unit: lbf*s
double trajectory_impulse = 50.0;  // unit: N*s

// BUG: adding lbf*s to N*s -- they have the same dimensions (momentum)
// but in unit-checker's dimensional model, both map to kg*m/s, so
// this won't be caught by dimensional analysis alone.
// However, if we treat them as distinct named units in a future version,
// this would be flagged.
double total_impulse = sm_forces_impulse + trajectory_impulse;

File: tests/fixtures/fortran/mars_climate_orbiter.f90
! Test fixture: Mars Climate Orbiter style mismatch.
!
! One module uses lbf*s (pound-force-seconds) for impulse,
! another expects N*s (newton-seconds).
! Both have the same dimensions (force * time = momentum),
! so this is a same-dimension test (no violation in dimensional
! analysis since lbf and N have same dimensions in our model).

program mars_climate_orbiter
  implicit none
  real :: thrust_impulse  ! unit: lbf*s
  real :: applied_force   ! unit: N
  real :: delta_v

  thrust_impulse = 4.45
  applied_force = 100.0
  delta_v = thrust_impulse + applied_force
end program mars_climate_orbiter
